File: String/3136. Valid Word/solution.swift
class Solution {
    func isValid(_ word: String) -> Bool {
        
        let vowels = Set(Array("aeiouAEIOU"))
        let uppercaseLetters = (65...90).map { Character(UnicodeScalar($0)!) }
        let lowercaseLetters = (97...122).map { Character(UnicodeScalar($0)!) }

        let consonants = Set(uppercaseLetters + lowercaseLetters).subtracting(vowels)
        
        let digits = Set(Array("0123456789"))
        
        var has_consonants = false
        var has_vowels = false
        
        if word.count < 3 {
            return false
        }
        
        
        for char in Array(word){
            
            if vowels.contains(char){
                has_vowels = true
            }
            else if consonants.contains(char){
                has_consonants = true
            }
            else if !digits.contains(char){
                return false
            }
        }
        
        return has_vowels && has_consonants
    }
}

let word = "234Adas"
print(Solution().isValid(word))

File: String/3136. Valid Word/solution.py
import string

class Solution:
    def isValid(self, word: str) -> bool:
        
        vowels = set('aeiouAEIOU')
        
        consonants = set(string.ascii_letters) - vowels

        digits = set('0123456789')

        has_vowels = False
        has_consonants = False
        

        if len(word) < 3:
            return False

        for c in word:
            if c in vowels:
                has_vowels = True
            elif c in consonants:
                has_consonants = True
            elif c not in digits:
                return False
            
        
        return has_vowels == has_consonants
            


if __name__ == '__main__':

    """ Example 1: """
    word = "234Adas"
    print(Solution().isValid(word))

    """ Example 2: """
    word = "3#H"
    print(Solution().isValid(word=word))